AutomatedTesting/Gem/PythonTests/Atom/tests/MaterialCanvas_Atom_BasicTests.py

```diff
@@ -123,7 +141,46 @@ def MaterialCanvas_BasicFunctionalityChecks_AllChecksPass():
             Tests.material_graph_name_is_test1,
             material_graph_name == "test1")
 
-        # 8. Look for errors and asserts.
+        # 8. Create a new material_graph from material_graph_document_ids[0].
+        material_graph = material_canvas_utils.get_graph(material_graph_document_ids[0])
+        material_graph_id = material_canvas_utils.get_graph_id(material_graph_document_ids[0])
+        Report.result(
+            Tests.material_graph_created,
+            material_graph.typename == "AZStd::shared_ptr<Graph>" and material_graph is not None)
+
+        # 9. Create a new world_position_node in memory.
+        world_position_node = material_canvas_utils.create_node_by_name(material_graph, "World Position")
+        Report.result(
+            Tests.world_position_node_created,
+            world_position_node.typename == "AZStd::shared_ptr<Node>" and world_position_node is not None)
+
+        # 10. Add the world_position_node to the material_graph.
+        # This test will be verified when the nodes are connected as if it doesn't exist the connection won't be made.
+        material_canvas_utils.add_node(material_graph_id, world_position_node, math.Vector2(-200.0, 10.0))
```

Comment:
we don't test that the node is on the graph. it could be an orphan node a pointer to something unconnected to anything. we need an ebus is_on_graph(AZStd::shared_pointer<Graph>, AZStd::shared_pointer<Node>)
if add_node fails how do we know is there an invalid node_id returned? is node_id again AZStd::shared_pointer<Node> or somethign else.
should this capture the return as `added_node` then validate that `added_node == world_position_node` since these seem like they should be equal return values.

Reply:
without getting the functions we need to retrieve node IDs we don't have an issue way to verify nodes on the graph other than testing that 2 nodes are created

this task seeks to solve that by adding a new function that gets the ID of a node when requested: https://github.com/o3de/o3de/issues/14517